### ISSUE TITLE
Permanent sorting order for Document List panel

### ISSUE THREAD

**Issue by mr-brunes** [notepad-plus-plus/notepad-plus-plus#11143]
### Description of the Issue

The Document List panel has headers for Name, Ext or Path etc. but when clickking on the header to set the list sorting method, it only sorts the document list once. This is counter intuitive with other apps where the column header indicates a permanent sort order and where one clicks on the column header to set it and/or change sort order direction (ascending/descending etc.) permanently. A single function is more suited to a button.

### Steps to Reproduce the Issue

1.  Open some documents and click "Name" header in the Document List panel to set alphabetical sorting
2.  Open another document
3.  Observe where the last docuemtn appears in the document list. 

### Expected Behavior

The last opened document shoud have appeared in alphabetical order (or whichever sort order has been chosen).

### Actual Behavior

The last document is most likely to be at the bottom rather than in alphabetical order (though this is unknown since the algorithm is nto pubclished in the manaul).

### Debug Information

Notepad++ v8.2.1   (64-bit)
Build time : Jan 19 2022 - 18:43:05
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : D:\Dropbox\Apps\Notepad++
OS Name : Windows 10 Pro (64-bit) 
OS Version : 2009
OS Build : 19044.1469
Current ANSI codepage : 1252
Plugins : AutoSave.dll mimeTools.dll NppConverter.dll NppExport.dll

**Comment by alankilborn** [comment]
>The last document is most likely to be at the bottom rather than in alphabetical order (though this is unknown since the algorithm is nto pubclished in the manaul).

There's no "algorithm".  Well I guess it is this:  A freshly opened file will appear as the "rightmost/bottommost" tab in a horizontal/vertical tab setup.

**Comment by mr-brunes** [comment]
Tx for the explanation, I guess it's an algorithm by default! Though I don't use tabs (not enough screen space) so have not been able to observe the behaviour. I use the Doument List as a vertical tab bar functionality replacement since the vertical tabs have vertical lettering)
Also I will edit my fat fingers typos!

**Comment by tiomanisland** [comment]
Not true! Nothing is auto sorted. If you keep opening new files, the list will not be sorted. SO many people have asked for tabs sorting or document list on the fly sorting options but all other people (devs?) do here is try to convince them they dont need auto sort. I guess it is extremely hard to add a checkbox  [x] autosort the list   [x] autosort tabs .... just say you dont know how to do it, simple.

**Comment by mr-brunes** [comment]
@tiomanisland Not sure what statement is being described as "not true". 
Also please can you give pointers to the other issues that have been raised concerning this issue with the responses as I searched first before filing this one and could not find anything?

All I can do is reiterate that the convention with other apps displaying lists with sorted columns is that the list is kept sorted each time the list contents are changed. With NPP the list sorting direction arrow remains when a new item is added to the list, but the list is no longer sorted. This makes the list sort direction arrow erroneous and confuses users trying to find an item expecting it to be in the list order as configured.
Finally an autosort checkbox should be not required since it is not the UX convention, just the implementation of the conventional sorting behaviour.

**Comment by tiomanisland** [comment]
I was referring to official Notepad++ forum (on their domain). Correct, auto-sorting should be implemented (checkbox in preferences maybe). People have been asking this for years and I wonder if NPP being developed by single person who hates sorting and how come someone can't add this simple feature?

**Comment by DigitalMenace242** [comment]
more stubborn weird thinking euro turds [VLC devs for an example] who do not have the brain cells to perceive simple logic.

**Comment by tiomanisland** [comment]
> more stubborn weird thinking euro turds [VLC devs for an example] who do not have the brain cells to perceive simple logic.

I really can't understand why a small tick box would be such a big problem.

**Comment by DigitalMenace242** [comment]
> > more stubborn weird thinking euro turds [VLC devs for an example] who do not have the brain cells to perceive simple logic.
> 
> I really can't understand why a small tick box would be such a big problem.

Boggles the mind..
even worse, I have to click on the doc list bar 3 times for the list to become a-z.

**Comment by mr-brunes** [comment]
I admit to being somewhat puzzled by the last few cryptic comments.
All I'm asking for is for the Document List to be autosorted.

To be clear, I'm not requesting a (Yet Another Option) checkbox for autosorting. Other apps and websites which display lists don't have a checkbox for this most basic of expected functionality i.e. for a list's sort settings to be sticky after setting by clicking on the column header and not to require constant user intervention after any list content change.

As noted before, the sort order direction indicator caret remains in place after a new item is added at the bottom, but the list is now in *unsorted* order.

In summary I'm just requesting that app follows UI convention.

**Comment by DigitalMenace242** [comment]
> I admit to being somewhat puzzled by the last few cryptic comments. All I'm asking for is for the Document List to be autosorted.
> 
> To be clear, I'm not requesting a (Yet Another Option) checkbox for autosorting. Other apps and websites which display lists don't have a checkbox for this most basic of expected functionality i.e. for a list's sort settings to be sticky after setting by clicking on the column header and not to require constant user intervention after any list content change.
> 
> As noted before, the sort order direction indicator caret remains in place after a new item is added at the bottom, but the list is now in _unsorted_ order.
> 
> In summary I'm just requesting that app follows UI convention.

That's all I want too. tired of opening new txt files and they go to the bottom on the list. I dont use tabs. just a doc list on the left side.

**Comment by CapnBrownShoes** [comment]
I would like to keep this request active.  Please, I deal with multiple documents (tabs) open at a given time which is incredible!  But  the feature to have that Document List populated automatically with a sort A-Z would be spectacular!

**Comment by MagratG** [comment]
I actually settle for just remembering my document list as I've sorted it!

**Comment by tiomanisland** [comment]
I don't understand why Notepad++ doesn't add CTRL+P as in VSCODE. This is the best option ever by far. I don't even know what tabs I have open, I just do CTRL+P and start typing. Enter and my file is here. 1 sec.

**Comment by Digita1Menace** [comment]
> I actually settle for just remembering my document list as I've sorted it!
Yea no shit. ridiculous.

**Comment by pryrt** [comment]
> I don't understand why Notepad++ doesn't add CTRL+P as in VSCODE. 

Because Notepad++ isn't VSCode?  Because no one has made an official request for it in Notepad++?  Because no one has submitted the code to do so?  Seems like there are a lot of possible reasons for it not to exist.  (I did a brief search, and couldn't find someone asking for specifically that, so I don't think that anyone has asked for it before.  So the answer to your statement really is "because no one has asked for it before", at least as far as I can find.)

If what you are saying is, "I would like it if Notepad++'s Document List feature had something akin to VSCode's `Ctrl+P` ["Quick Open" feature](https://code.visualstudio.com/docs/editing/editingevolved#_quick-file-navigation), whereby, in the Document List, you could search for / filter by specific text in the file/tab names", that would be an excellent feature request.  So, instead of doing a comment in a semi-related (but separate) feature request, your request will only get the visibility and tracking you would desire if you create a separate issue asking for that specific feature.  

In the Community Forum recently, [someone asked about](https://community.notepad-plus-plus.org/topic/27430/search-in-open-tabs-not-dropdown) a similar action be put in the Tab Bar right-click context menu (which would be another excellent place for it).  An alternative mentioned in that discussion is using the Explorer Plugin, which has a way to search the filenames/tab-names.

But if that plugin doesn't quite give you what you want, then I would recommend starting your own Feature Request issue, rather than just burying your request as a comment on a separate request.  (Having the ability to search the Document List and/or Tab Bar would not "close" a request to be able to have permanent sort order for Document List, so it's a separate feature request.)

**Comment by tiomanisland** [comment]
pryrt sorry for that. I don't even know if someone would write such "extension". I remember I asked for one small feature long time ago but what I got was "write it yourself". Ok, I can't do that.  I just wanted to say that I have been using Notepad++ for 20 years (I stopped last year just because of CTRL+P missing). I tried plugins but none of them did that. And most of them are messy, some look like crap from 1980. Some tried to do that but they werent able to look in subfolders or they needed 10 seconds each time to load the list LOL. If someone know how to do CTRL+P - just that - N++ would be much better.

**Comment by pryrt** [comment]
I won't reply to the rest of your statements (despite having a lot to say).  But I will reply to this

> If someone know how to do CTRL+P - just that - N++ would be much better.

As I said before, if you want the feature, you need to make a new feature request.  This comment on a separate issue does _not_ count as a feature request.
